src/Hamiltonians/TwoBodyRealCorrelation.jl

```diff
@@ -0,0 +1,60 @@
+"""
+    G2RealCorrelator(d::Int) <: AbstractHamiltonian{Float64}
+
+Two-body operator for density-density correlation between sites separated by `d`
+with `0 ≤ d < M`. Can be applied to a vector in any [`SingleComponentFockAddress`](@ref) basis.
+```math
+    \\hat{G}^{(2)}(d) = \\frac{1}{M} \\sum_i^M \\hat{n}_i (\\hat{n}_{i+d} - \\delta_{0d}).
+```
+Assumes periodic boundary conditions where
+```math
+    \\hat{G}^{(2)}(-M/2 \\leq d < 0) = \\hat{G}^{(2)}(|d|),
+```
+```math
+    \\hat{G}^{(2)}(M/2 < d < M) = \\hat{G}^{(2)}(M - d),
+```
+and normalisation
+```math
+    \\sum_{d=0}^{M-1} \\langle \\hat{G}^{(2)}(d) \\rangle = \\frac{N (N-1)}{M}.
+```
+
+# Arguments
+- `d::Integer`: distance between sites.
+
+# See also
+
+* [`HubbardReal1D`](@ref)
+* [`G2MomCorrelator`](@ref)
+* [`AbstractHamiltonian`](@ref)
```

Comment:
It would be good to also add a link to `AllOverlaps` here, as this is the context where this correlator will be used. Please also do the same for `G2MomCorrelator`.